File: Java implementation/src/Operation.java
import java.util.HashSet;
import java.util.Set;

public class Operation implements Arithmetic {

    private final String operation;
    private final Arithmetic a1;
    private final Arithmetic a2;

    public Operation(String operation, Arithmetic a1, Arithmetic a2) {
        this.operation = operation;
        this.a1 = a1;
        this.a2 = a2;
    }

    @Override
    public Set<String> fv() {
        Set<String> fv = new HashSet<>();
        fv.addAll(a1.fv());
        fv.addAll(a2.fv());
        return fv;
    }

    @Override
    public String toString() {
        return a1.toString() +" "+ operation +" "+ a2.toString();
    }

    ;


}

File: CppUpdated/BasicActions/Arithmetics/Operation.cpp
//
//  Operation.cpp
//  ProgramAnalysis
//
//  Created by Sahas Arora on 11/27/20.
//  Copyright © 2020 Sahas Arora. All rights reserved.
//

#include "Operation.h"

Operation::Operation(string operation, Arithmetic a1, Arithmetic a2) {
    this->op = operation;
    this->a1 = a1;
    this->a2 = a2;
}

Identifiers Operation::getUsedVariables() {
    return a1.getUsedVariables().merge(a2.getUsedVariables());
}

string Operation::toString() {
    return a1.toString() + op + a2.toString();
}